### ISSUE TITLE
High memory usage

### ISSUE THREAD

**Issue by lufinkey** [lufinkey/MobileBrawler#7]
The new version of iSSB uses almost 4 times the amount of memory that its predecessor did. This is almost undoubtedly because of the new image resources being used. I need to find a way to lower down the memory usage of these resources.
(this is low priority for now, until I start moving everything over to iOS)

**Comment by darkroastcreative** [comment]
Image resources being image files?

**Comment by lufinkey** [comment]
Yes. The new images in the project, when they're loaded into memory, take
up waaaay more space than the previous version. I've reduced the load to
50mb, but my goal is 25mb like the original version of iSSB.

**Comment by lufinkey** [comment]
The unfortunate part is that this can't be fixed by compressing the images,
since the images are fully loaded into memory as RGBA arrays. So the images
themselves just need to be sized down.

**Comment by Kfeavel** [comment]
@lufinkey That or load them all at launch and use caches in order to reduce memory and loading lag. You'll have a memory spike at launch (the ~50mb), but then if you cache the ones which aren't used often on the disk (characters and stages) and the ones used really often (menus and such) in memory, then you should be able to bring that down a bit more towards your ~25mb goal. Also, if you dump the caches from memory onto the disk on exit (in the caches or temp folder) then the only times you'll have that initial launch spike is when iOS (or Windows) cleaned the cache, which shouldn't be often.
